src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -404,22 +411,39 @@ vector<SparqlTripleSimple> Visitor::visit(Parser::DeleteDataContext* ctx) {
 }
 
 // ____________________________________________________________________________________
-void Visitor::visit(Parser::DeleteWhereContext* ctx) {
-  // DeleteData only deletes the specified triples. DeleteWhere matches the
-  // given pattern and then deletes the matched triples. This is a shortcut for
-  // a modify query.
-  reportNotSupported(ctx, "SPARQL 1.1 Update DeleteWhere is not supported.");
+std::pair<vector<SparqlTripleSimple>, ParsedQuery::GraphPattern> Visitor::visit(
+    Parser::DeleteWhereContext* ctx) {
+  auto triples = visit(ctx->quadPattern());
+  auto transformTriple =
```

Comment:
This function looks like it should be
`SparqlTripleSimple::toSparqlTriple() const`
or `static SparqlTriple::fromSimple()`
(depending on which one is easier b.c. of cyclic dependencies.

Reply:
I have placed it in `SparqlTriple` for stylistic reasons. If one goes purely after the design there could be no conversion between the two types. One is a PropertyPath and the other is many things but a PropertyPath. The PropertyPath still accepts values that would have other types like Variables or Iris. I think that it is better to keep this weirdness in the type that uses PropertyPath instead of letting it leak into an unrelated type.